WooCommerce/Classes/POS/Presentation/CartView.swift

```diff
@@ -47,6 +47,7 @@ struct CartView: View {
 
                     Button {
                         posModel.removeAllItemsFromCart()
+                        ServiceLocator.analytics.track(.pointOfSaleClearCartTapped)
```

Comment:
I don't love that we use `ServiceLocator` within the view (not just here). Of course, if/until we decide to decouple POS from the main application that wouldn't change anything. I guess that wouldn't be a hard refactoring, we would need to inject dependencies through the environment.

Reply:
Yeah, I don't like to see the ServiceLocator being called on those neither, but as you mention the potential refactor would be minimal. 

There are several views that are only driven by the aggregate model, so one of my thoughts around this was if was worth to expose either the analytics dependency in the aggregate model (make it just `private(set)`) or have some `track(Event)` function in the aggregate where we pass all of those and handle events there through the internal dependency. One of the reasons I didn't is we don't solve anything regarding unit testing in most cases, but might be alternatives indeed. Knowing for sure if we're going to decouple from the app or not would have definitely helped in a lot of these decisions 😅